src/DSHARP_opacs.cu

```diff
@@ -47,6 +47,28 @@ __global__ void _calc_rho_kappa(GridRef g, Field3DConstRef<Prims> qd, FieldConst
     }
 } 
 
+__global__ void _calc_rho_kappa(GridRef g, Field3DConstRef<double> rho_d, FieldConstRef<Prims> wg, 
+                                DSHARP_opacsRef opacs, Field3DRef<double> rhokabs, Field3DRef<double> rhoksca) {
+
+    int k = threadIdx.x + blockIdx.x*blockDim.x ;
+    int j = threadIdx.y + blockIdx.y*blockDim.y ;
+    int i = threadIdx.z + blockIdx.z*blockDim.z ;
+
+    if (k < opacs.n_lam && j < g.Nphi + 2*g.Nghost && i < g.NR+2*g.Nghost) {
+
+        double rhok_dust_abs = 0;
+        double rhok_dust_sca = 0;
+
+        for (int l=0; l<opacs.n_a; l++) { 
+            rhok_dust_abs += rho_d(i,j,l)*opacs.k_abs(l,k);
+            rhok_dust_sca += rho_d(i,j,l)*opacs.k_sca(l,k);
+        }
+
+        rhokabs(i,j,k) = wg(i,j).rho*opacs.k_abs_g(k) + rhok_dust_abs;
+        rhoksca(i,j,k) = wg(i,j).rho*opacs.k_sca_g(k) + rhok_dust_sca;
+    }
+} 
+
```

Comment:
Is there a reason this function does not also compute rho_tot? I thought rho_tot was needed for some of the radiation routines. This seems like a natural place to do it.

Reply:
Yeah good point, will make these changes now.